assisted_service_mcp/src/utils/log_analyzer/log_analyzer.py

```diff
@@ -131,6 +138,66 @@ def get_events_by_host(self) -> Dict[str, List[Dict[str, Any]]]:
                 events_by_host[event["host_id"]].append(event)
         return events_by_host
 
+    @staticmethod
+    def get_hostname(host: Dict[str, Any]) -> str:
+        """Extract hostname from host metadata."""
+        hostname = host.get("requested_hostname")
+        if hostname:
+            return hostname
+
+        try:
+            inventory = json.loads(host["inventory"])
+            return inventory["hostname"]
+        except (KeyError, json.JSONDecodeError):
+            return host.get("id", "unknown")
+
+
+class LogAnalyzer(ClusterAnalyzer):
+    """Analyzer for OpenShift Assisted Installer logs."""
+
+    _metadata: dict[str, Any] | None
+
+    def __init__(self, logs_archive: nestedarchive.RemoteNestedArchive):
+        """
+        Initialize the log analyzer.
+
+        Args:
+            logs_archive: RemoteNestedArchive containing the cluster logs
+        """
+        super().__init__()
+        self.logs_archive = logs_archive
+
+    @property
+    def metadata(self) -> Dict[str, Any] | None:
+        """Get cluster metadata."""
+        if self._metadata is None:
+            try:
+                metadata_content = self.logs_archive.get("cluster_metadata.json")
```

Comment:
It's a bit strange to me that we now sometimes get the metadata from the API and other times get it from the log bundle.

Did you consider making this consistent (I guess just always getting it from the API)?

I could see a situation where if the logs are available we're actually acting on old data where if they are not then we get the data from the live cluster.

Also it's a bit odd that the user of the `ClusterAnalyzer` is responsible for fetching the metadata and events from outside the object and setting them where `LogAnalyzer` handles that for the caller. We should probably be more consistent there as well.